### ISSUE TITLE
[13.0] Base: _filter_access_rules / _filter_access_rules_python

### ISSUE THREAD

**Issue by keshrath** [odoo/odoo#38570]
Impacted versions:

13.0

Steps to reproduce:

```python
records = self.browse([1, 2, 3]) # No access on record with ID 1

records._filter_access_rules("read").ids >> [2, 3]
records._filter_access_rules_python("read").ids >> AccessError
```

Current behavior:

The method `_filter_access_rules_python ` is now used in `check_access_rule` instead of `_filter_access_rules ` and there seem to be some cases where this causes an AccessError instead of filtering the records.

I think its caused by the mapped function since it tries to read the field while having no access to it.

- https://github.com/odoo/odoo/blob/13.0/odoo/models.py#L5123

Expected behavior:

Filter the records instead of throwing an error.

**Comment by rco-odoo** [comment]
Dear @keshrath, this issue has been fixed by cff5cc8cc78a32eb204368e84ec4980324046405.

**Comment by r13ssh** [comment]
We are currently on the 13.0 branch commit # 6ecc34641f0f7c378db5cc15ddaddb63e1ccbd6f (i.e. above referenced commit is part of our codebase) running Odoo CE and seem to be hitting this scenario again, in our case in a `create` scenario.

As in the above case, `filter_access_rules_python` does not return the new records (the `create` hence fails), while `fetch_access_rules` does (correctly) return them.

Please let me know if I you would like me to provide additional details.
